src/DataTypes/Serializations/SerializationNumber.cpp

```diff
@@ -152,6 +166,15 @@ void SerializationNumber<T>::deserializeBinaryBulk(IColumn & column, ReadBuffer
     size_t initial_size = x.size();
     x.resize(initial_size + limit);
     size_t size = istr.readBig(reinterpret_cast<char*>(&x[initial_size]), sizeof(typename ColumnVector<T>::ValueType) * limit);
+    if constexpr (std::endian::native == std::endian::big && sizeof(T) >= 2)
+    {
+        for (size_t i = 0; i < limit; i++)
+        {
+            char *start = reinterpret_cast<char*>(&x[initial_size + i]);
+            char *end = start + sizeof(typename ColumnVector<T>::ValueType);
+            std::reverse(start, end);
+        }
```

Comment:
`transformEndianness` should help you out here as well.

Reply:
Done, used your latest code, reverted this change.